odoo/addons/test_new_api/tests/test_onchange.py

```diff
@@ -329,7 +330,12 @@ def test_onchange_specific(self):
         discussion = self.env.ref('test_new_api.discussion_0')
         demo = self.user_demo
 
-        field_onchange = self.Discussion._onchange_spec()
+        view_info = self.Discussion.get_view()
+        tree = etree.fromstring(view_info['arch'])
+        field_participants = tree.xpath('//field[@name="participants"]')[0]
+        field_participants.remove(field_participants.xpath('./form')[0])
+        view_info['arch'] = etree.tostring(tree)
+        field_onchange = self.Discussion._onchange_spec(view_info=view_info)
```

Comment:
Why?

Reply:
I checked a little more

The behavior of `onchange` changes when the form is added in addition to the tree view, even if the form view is not visible in the interface, only the tree is visible.

And that, in the web client as well.
In current master, without embedding the `<form>` view in the `<field name="participants"></field>`:
![image](https://user-images.githubusercontent.com/5822488/163186815-6acf0862-d132-4415-980d-cc7a86ea3253.png)

In current master, embedding the default `<form>` view of `res.users` in `<field name="participants"></field>` (manually, by editing the view to add the default user form to `<field name="participants"></field>`)
![image](https://user-images.githubusercontent.com/5822488/163187006-939cf0e2-8593-46ef-ae56-09c9433ce62c.png)

When the form is embedded in the view, despite there is only the tree visible in the web interface, the `field_onchange` sent to `def onchange(self, values, field_name, field_onchange):` changes to add the fields from the form view, and this changes the result, even in the web client. The `participants` commands list goes from a clear + links commands to a clear + set commands.

To be honest I do not fully understand why there is this behavior.

Another possible diff for this test would be:
```diff
diff --git a/odoo/addons/test_new_api/tests/test_onchange.py b/odoo/addons/test_new_api/tests/test_onchange.py
index c0e115bc7048..48cce1519972 100644
--- a/odoo/addons/test_new_api/tests/test_onchange.py
+++ b/odoo/addons/test_new_api/tests/test_onchange.py
@@ -333,7 +333,10 @@ class TestOnChange(SavepointCaseWithUserDemo):
         self.assertEqual(field_onchange.get('moderator'), '1')
         self.assertItemsEqual(
             strip_prefix('participants.', field_onchange),
-            ['display_name'],
+            [
+                'display_name', 'id', 'avatar_128', 'image_1920', 'name',
+                'email', 'login', 'company_id', 'groups_id', 'phone', 'mobile'
+            ],
         )

         # first remove demo user from participants
@@ -353,8 +356,8 @@ class TestOnChange(SavepointCaseWithUserDemo):

         self.assertIn('participants', result['value'])
         self.assertItemsEqual(
-            result['value']['participants'],
-            [Command.clear()] + [Command.link(user.id) for user in discussion.participants + demo],
+            [participant[0:2] for participant in result['value']['participants']],
+            [Command.clear()[0:2]] + [Command.update(user.id, {})[0:2] for user in discussion.participants + demo],
         )

     def test_onchange_default(self):
```

Reply:
IMO the form view in the field `participants` is mostly useless, and can even be considered harmful.
What about using widget `many2many_tags` for that field instead?

Reply:
That is the best idea ... Thanks !